frontend-zs/src/App.js

```diff
@@ -36,14 +37,15 @@ class App extends React.Component {
     return (
       <UserContext.Provider>
         <Router>
-          <Header/>
+          <Header />
           <Switch>
             <Route path="/editProfile" exact component={EditProfile} />
             <Route path='/login' component={Login} />
             <Route path='/signup' component={Signup} />
             <Route path="/registrationStepTwo" component={RegistrationStepTwo} />
             <PrivateRoute path='/projects' component={ProjectsList} />
-            <PrivateRoute path="/stories" component={StoryList} />
+            <PrivateRoute path='/stories' component={StoryList} />
+            <PrivateRoute path='/story/:id' component={StoryPage} />
```

Comment:
```suggestion
            <PrivateRoute path='/story/:id' component={StoryPage} />
            <PrivateRoute path='/stories' component={StoryList} />
```

Reply:
The problem with PrivateRoute?

Reply:
yes

Reply:
It doesn't solve the problem. I switched the routes and change my fix in PrivateRoute.js. The id is undefined in StoryPage.js and the data can not be loaded from the db.

Reply:
I found this solution here: https://medium.com/@thanhbinh.tran93/private-route-public-route-and-restricted-route-with-react-router-d50b27c15f5e